File: student-grading-system-web/src/types/assignment.ts
export interface Assignment {
  id: number;
  title: string;
  description: string;
  dueDate: string;
  maxMarks: number;
  subjectName: string;
  createdAt: string;
}

export interface CreateAssignmentRequest {
  title: string;
  description: string;
  subjectId: number;
  dueDate: string;
  maxMarks: number;
}

export interface UpdateAssignmentRequest {
  title: string;
  description: string;
  dueDate: string;
  maxMarks: number;
}

File: StudentGradingSystem.Api/Models/Assignment.cs
using System.ComponentModel.DataAnnotations;

namespace StudentGradingSystem.Api.Models;

public class Assignment
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(1000)]
    public string Description { get; set; } = string.Empty;

    public int SubjectId { get; set; }
    public Subject Subject { get; set; } = null!;

    public int FacultyId { get; set; }
    public Faculty Faculty { get; set; } = null!;

    public DateOnly DueDate { get; set; }

    public decimal MaxMarks { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<AssignmentSubmission> Submissions { get; set; }
    = new List<AssignmentSubmission>();
}